### PR TITLE
Add backend logic to Skilltree UI

### PR DESCRIPTION
**Pull request by veteran29**
- title

### CODE REVIEW

**Comment by veteran29** [comment]
It has some looks issues since the recent merge of the UI changes to dev but they weren't there before so they can be deemed unrelated to the logic.

File: game/functions/core/client/displays/fn_displaySkills.sqf [hunk 1]
```diff
@@ -1,70 +1,96 @@
...
 params ["_mode", "_params"];
+_this = _params;
+
 switch _mode do {
-    case "onLoad":{
-        _params params ["_display"];
+    case "onLoad": {
+        params ["_display"];
+
+        vgm_skills_ui_currentSkillTree = createHashMap;
```
Comment (7erra) [inline game/functions/core/client/displays/fn_displaySkills.sqf]: Are global variables necessary or could they be local to the display with `_display setVariable`?

File: game/functions/systems/skills/client/fn_skills_openSkillTree.sqf [hunk 2]
```diff
@@ -19,218 +18,4 @@
...
-};
-
-// ensure that we display tabs in config order, hashmaps are unordered
-private _skillTreeClasses = "true" configClasses (missionConfigFile >> "vgm_skillTrees") apply {configName _x};
-
-private _skillTreesHash = missionNamespace getVariable "vgm_skills_treesHashMap";
-{
-    [_display, _skillTreesHash get _x, _forEachIndex] call _fnc_draw;
-} forEach _skillTreeClasses;
+findDisplay 46 createDisplay "VGM_DisplaySkills";
```
Comment (7erra) [inline game/functions/systems/skills/client/fn_skills_openSkillTree.sqf]: `createDialog` would be better unless you need the reference to the display as `createDialog` removes the ability to move around

File: game/functions/core/client/displays/fn_displaySkills.sqf (hunk 1 above)
Comment (veteran29) [reply game/functions/core/client/displays/fn_displaySkills.sqf]: I'll switch to display vars. Less to clear in `onUnload`

File: game/functions/core/client/displays/fn_displaySkills.sqf [hunk 3]
```diff
@@ -1,70 +1,94 @@
...
+            ["refreshUI", _display] call vgm_c_fnc_displaySkills;
+        }]] call para_g_fnc_event_subscribeLocal;
+        _display setVariable ["vgm_skills_ui_learntHandlerId", _handlerId];
+
+        ["refreshUI", _display] call vgm_c_fnc_displaySkills;
+    };
+
+    case "onUnload": {
+        params ["_display"];
+        systemChat str _display;
```
Comment (Spoffy) [inline game/functions/core/client/displays/fn_displaySkills.sqf]: Is this needed for debugging or can we remove? :)

Comment (veteran29) [reply game/functions/core/client/displays/fn_displaySkills.sqf]: leftover. gotta remove.

**Pull request merged**